backbone/src/main/java/org/researchstack/backbone/ui/ConsentTaskActivity.java

```diff
@@ -49,12 +51,19 @@
     private String signatureBase64;
 
     public static Intent newIntent(Context context, Task task, String assetsFolder) {
-        Intent intent = new Intent(context, ConsentViewTaskActivity.class);
+        Intent intent = new Intent(context, ConsentTaskActivity.class);
         intent.putExtra(EXTRA_TASK, task);
         intent.putExtra(EXTRA_ASSETS_FOLDER, assetsFolder);
         return intent;
     }
 
+    @Override
+    protected void onCreate(@org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        getViewModel().setStepCallbacks(this);
```

Comment:
You shouldn't do this, I mean, make the view model hold a reference to the activity. This will cause memory leaks.
If you need some sort of communication between the View/Fragment/activity you can use the view model directly

Note: I would also convert ConsentTaskActivity to kotlin, and use koin to inject the viewModel

Reply:
that makes more sense now that you mention it, ok I'll do that instead. Thanks :)